x-pack/platform/plugins/shared/stack_connectors/server/connector_types/inference/inference.ts

```diff
@@ -128,11 +128,13 @@ export class InferenceConnector extends SubActionConnector<Config, Secrets> {
     const obs$ = from(eventSourceStreamIntoObservable(res as unknown as Readable)).pipe(
       filter((line) => !!line && line !== '[DONE]'),
       map((line) => {
-        return JSON.parse(line) as OpenAI.ChatCompletionChunk | { error: { message: string } };
+        return JSON.parse(line) as
+          | OpenAI.ChatCompletionChunk
+          | { error: { message?: string; reason?: string } };
       }),
       tap((line) => {
         if ('error' in line) {
-          throw new Error(line.error.message);
+          throw new Error(line.error.message ?? line.error.reason ?? 'Unknown error');
```

Comment:
I suggest using `||` instead of `??`, as `??` will allow `""` to pass through as the value:

```console
> x = ""
''
> y = ""
''
> x ?? y ?? 'abc'
''
> x || y || 'abc'
'abc'
```